src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -611,14 +605,16 @@ private string GenerateExtendedAttributeName()
         }
 
         // Gets the special name for the 'name' field in a global extended attribute entry.
-        // Format: "%d/GlobalHead.%p/%f"
+        // Format: "%d/GlobalHead.%p/%n"
         // - %d: The path of the $TMPDIR variable, if found. Otherwise, the value is '/tmp'.
         // - %p: The current process ID.
         // - %n: The sequence number of the global extended header record of the archive, starting at 1. In our case, since we only generate one, the value is always 1.
         // If the path of $TMPDIR makes the final string too long to fit in the 'name' field,
         // then the TMPDIR='/tmp' is used.
-        private static string GenerateGlobalExtendedAttributeName()
+        private static string GenerateGlobalExtendedAttributeName(int globalExtendedAttributesEntryNumber)
         {
+            Debug.Assert(globalExtendedAttributesEntryNumber >= 1);
+
```

Comment:
As an aside - the code below reading TMPDIR then falling back to /temp- why doesn't it just call Path.GetTempPath() that does the same thing?

Reply:
Good suggestion. I'll use `Path.GetTempPath`.

Here's why I wasn't using that:

The [GNU tar manual](https://www.gnu.org/software/tar/manual/html_section/Portability.html) specifies how the global extended attribute header name field obtains its value:

> ### 8.3.7.1 Controlling Extended Header Keywords
>
> `$TMPDIR/GlobalHead.%p.%n`
>
> [...] `$TMPDIR` stands for the value of the `TMPDIR` environment variable. If `TMPDIR` is not set, tar uses `/tmp`.

This description looks very similar to the remarks section of [Path.GetTempPath](https://docs.microsoft.com/en-us/dotnet/api/system.io.path.gettemppath?view=net-6.0&tabs=linux#remarks):

> ### Remarks
> 
> This method checks for the existence of environment variables in the following order and uses the first path found:
>
> Linux
>
> 1. The path specified by the TMPDIR environment variable. If the path is not specified in the TMPDIR environment variable, the default path /tmp/ is used.
>
> Windows
> 
> 1. The path specified by the TMP environment variable.
> 2. The path specified by the TEMP environment variable.
> 3. The path specified by the USERPROFILE environment variable.
> 4. The Windows directory.

So a couple of things need to be considered if we are to use `Path.GetTempPath`:

- I don't see why any tool would be reading the GEA path, or having reading/writing behavior depend on it. The field isn't very interesting, unless the user cares about the process ID or the sequence number.
- Keep in mind that none of the tar specs (at least the ones I read) specify what the expected behavior for Windows should be. Everything is Unix focused. If we format that string embedding Windows paths, I hope other tools don't break due to finding unexpected drive names or `\` separators in that field. But go back to the previous point: I don't think any tool should make its behavior depend on this field.
- If the Windows path ends up being larger than than 100 bytes, the sequence number and process ID would get truncated. Which is fine, I guess, because a couple of lines below, I re-generate the string by forcing the usage of `/tmp`.